SPECS/golang/golang.spec

```diff
@@ -13,20 +15,20 @@
 %define __find_requires %{nil}
 Summary:        Go
 Name:           golang
-Version:        1.20.10
+Version:        1.21.6
 Release:        1%{?dist}
 License:        BSD-3-Clause
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 Group:          System Environment/Security
-URL:            https://golang.org
-Source0:        https://golang.org/dl/go%{version}.src.tar.gz
+URL:            https://github.com/microsoft/go
+Source0:        https://github.com/microsoft/go/releases/download/v%{version}-%{ms_go_revision}/go.%{ms_go_buildid}.src.tar.gz
 Source1:        https://dl.google.com/go/go1.4-bootstrap-20171003.tar.gz
```

Comment:
Maybe worth mentioning that we are also hosting this at https://github.com/microsoft/go/releases/tag/v1.4.0-1, but I don't see any pressing reason to change the URL in this spec. The checksum is the same, and a download cache is used to actually build the package.

Reply:
If it's the same, I think it's still worth updating the URL so that there's no confusion about why we're using partially Microsoft and partially Google's sources. And if the name of the file is different in Microsoft's URL, adding `#<cached_file_name>` at the end of the URL will make the tool still pick the right file from our cache.

Reply:
Sure, pushed. Thanks for the note about `#`, that might be handy if we end up hitting some wall in our automation.

In theory, it might be useful to keep upstream's URL to show people that this package does use the same bootstrap sources as ordinary Go distributions. So, added a comment, too.